Tmedia 729 subheadline

JackHowa opened the pull request:
## Description

Update subheadline to always render h2 and use new architecture

## Jira Ticket

- [TMEDIA-729](https://arcpublishing.atlassian.net/browse/TMEDIA-729)

## Acceptance Criteria

Headline Block

Subheadline block is migrated to the new architecture 

Includes Custom fields of:

‘Value to display’ - with dropdown options of Subheadline or Description

## Test Steps

1. Checkout this branch `git checkout TMEDIA-729-subheadline`
2. `npm run storybook`
3. Ensure this matches the heading level 6 documentation in the storybook
![Screen Shot 2022-05-16 at 14 01 54](https://user-images.githubusercontent.com/5950956/168663722-a44faac2-c19f-40a8-a235-32c5a7d10959.png)

## Effect Of Changes

### Before

- uses font-size 20 and 17.6px in live environment for mobile and desktop, respectively

### After

- Now it uses matching heading level 6 based on migration guidance 

## Dependencies or Side Effects

- Other prs for manifest and feature pack 
- https://github.com/WPMedia/arc-themes-manifests/pull/52 https://github.com/WPMedia/arc-themes-feature-pack/pull/244

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

# Question: 

- There's no matching global font-size for 26px desktop size. What is the desired resolution for these cases? Referenced in ticket -> using header 6 based on design question in other pr
- Removed what looked like test code that leaked in of setting the default value to be "Subheadline". I removed this. This seems like it would be ok, right? We don't have this pattern elsewhere. -> it is needed

JackHowa committed:
feat: Add subheadline features for v2 architecture

JackHowa committed:
fix: Seems to be test code that leaked in. Removing to be empty value by default

Commit:
Merge branch 'arc-themes-release-version-2.00' into TMEDIA-729-subheadline

nschubach commented on blocks/subheadline-block/package.json:
```diff
@@ -18,15 +19,8 @@
   },
   "gitHead": "83fb8d6685958d85ddbfcfbf01d9a5c864cd6c95"
-}
+}
```
```suggestion
}

```

nschubach commented on blocks/subheadline-block/features/subheadline/default.test.jsx:
```diff
@@ -19,7 +19,7 @@ describe("the subheadline feature for the default output type", () => {
 		it("should dangerously set the innerHTML to the subheadline content", () => {
 			const wrapper = mount(<SubHeadline />);
-			expect(wrapper.find("h2.sub-headline").at(0).prop("dangerouslySetInnerHTML")).toStrictEqual({
+			expect(wrapper.find("h2.b-subheadline").at(0).prop("dangerouslySetInnerHTML")).toStrictEqual({
```
I think for Testing Library standard, it should be something like (not tested, just going by memory here):
```suggestion
			expect(wrapper.text()).toBe("subheadline for our story");
```
which removes the need to refactor tests if the class name or header hierarchy changes.  Unless what we are testing IS the hierarchy which is a concern for the HeadingSection

nschubach replied:
Then this could be a `shallow` render if I'm not mistaken.

JackHowa replied:
I don't think the h hierarchy is intended to be tested. I updated the classname based on using the architecture

matthewroach replied:
What was the reason for resolving this comment? The tests would be less brittle by changing to the way @nschubach suggested and testing the text value of the component

JackHowa replied:
Not sure I follow. I wanted to have previous testing parity by updating the classname based on the albeit brittle structure of the previous tests. I've updated to test the entire container if that's the suggestion as I understand it?

JackHowa replied:
updated @matthewroach @nschubach based on using `text()` on the whole container.

nschubach commented on blocks/subheadline-block/features/subheadline/default.jsx:
```diff
@@ -18,15 +15,9 @@ export const SubHeadlinePresentation = ({
-		)
-	);
+	return value ? (
+		<h2 className="b-subheadline" dangerouslySetInnerHTML={{ __html: value }} />
```
HeadingSection > Heading ?

JackHowa replied:
This will always render as an h2 similar to headline only using h1, right? only the header is using heading/heading section based on my my reading https://arcpublishing.atlassian.net/jira/software/c/projects/TMEDIA/boards/875?modal=detail&selectedIssue=TMEDIA-712

JackHowa replied:
@matthewroach what do you think? 

Is subheadline always going to be an h2? or will it use the heading section/heading architecture? I was using the h2 always as the headline it’s commonly used with will always be an h1 and does not use the heading section component either

brentmiller1973 replied:
Curious about @JackHowa comment above.  Only thing imo holding this ticket up

matthewroach replied:
Switching to HeadingSection > Heading would give the same output by default - as this block relies on global content it does have a limited use case on where it can be placed and ideally it would come after the Headline component. For consistency I think we should update it to use the HeadingSection > Heading set up

JackHowa replied:
updated @matthewroach @nschubach -- thanks

JackHowa committed:
feat: Add font-weight and subheadline corresponding styles to line height and size

JackHowa committed:
feat: Use heading level 6 for subheadline

JackHowa committed:
Merge branch 'TMEDIA-729-subheadline' of github.com:WPMedia/fusion-news-theme-blocks into TMEDIA-729-subheadline

Commit:
Update blocks/subheadline-block/package.json

Co-authored-by: nschubach <nschubach@gmail.com>

JackHowa committed:
fix: Revert change for subheadine default value. Not used as a value

JackHowa committed:
Merge branch 'TMEDIA-729-subheadline' of github.com:WPMedia/fusion-news-theme-blocks into TMEDIA-729-subheadline

JackHowa committed:
revert: Add back deleted prop types peer dep

nschubach commented on .storybook/themes/news.scss:
```diff
@@ -215,6 +224,11 @@
 					grid-area: main-right-rail-bottom,
 				),
+				subheadline: (
+					font-size: var(--global-font-size-5),
+					line-height: var(--global-line-height-4),
+					font-weight: var(--global-font-weight-7),
```
```suggestion
					font-size: var(--global-font-size-5),
					font-weight: var(--global-font-weight-7),
					line-height: var(--global-line-height-4),
```

nschubach commented on .storybook/themes/news.scss:
```diff
@@ -312,6 +326,11 @@
 					padding: 0 var(--global-spacing-6) 0 0,
 					display: flex,
 				),
+				subheadline: (
+					font-size: var(--global-font-size-7),
+					line-height: var(--global-line-height-9),
+					font-weight: var(--global-font-weight-7),
```
```suggestion
					font-size: var(--global-font-size-7),
					font-weight: var(--global-font-weight-7),
					line-height: var(--global-line-height-9),
```

nschubach replied:
Seems to be a discrepancy here that I don't see being called out?

Gazette

corecomponents subheadline: (line-height: 2rem)
![image](https://user-images.githubusercontent.com/2287238/168881836-7a2a77cb-55f3-42b7-97a5-0776730d3752.png)

vs localhost with feature pack branch for this: (line-height: 3.5rem)
![image](https://user-images.githubusercontent.com/2287238/168881984-eecd9399-2991-4aa2-b8c7-f9549f561f10.png)

JackHowa replied:
I'm going off of https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3138945094/Blocks+Design+Discrepancies discussion to match the typography with the new scale. I can ask if the line height should be included in the change if we want

matthewroach replied:
Looks like the Figma token documentation for the line height is referencing the wrong token name it should be `line-height-5`

JackHowa replied:
will update

JackHowa replied:
updated!

nschubach reviewed (commented):
I could be missing something...

Commit:
Merge branch 'arc-themes-release-version-2.00' into TMEDIA-729-subheadline

Commit:
Merge branch 'arc-themes-release-version-2.00' into TMEDIA-729-subheadline

JackHowa committed:
fix: Remove duplicate font-weight

JackHowa committed:
fix: Use heading subsection per feedback

Via https://github.com/WPMedia/arc-themes-blocks/pull/1346#discussion_r875707602

Commit:
Update .storybook/themes/news.scss

Co-authored-by: nschubach <nschubach@gmail.com>

Commit:
Update .storybook/themes/news.scss

Co-authored-by: nschubach <nschubach@gmail.com>

JackHowa committed:
fix: Update style per figma with mocks out of date

Commit:
Merge branch 'arc-themes-release-version-2.00' into TMEDIA-729-subheadline

JackHowa committed:
fix: Update text to look at the entire container

JackHowa committed:
Merge branch 'TMEDIA-729-subheadline' of github.com:WPMedia/fusion-news-theme-blocks into TMEDIA-729-subheadline
